lightning/src/chain/chainmonitor.rs

```diff
@@ -94,11 +94,11 @@ where C::Target: chain::Filter,
 	/// [`ChannelMonitor::transactions_confirmed`].
 	///
 	/// Used instead of [`block_connected`] by clients that are notified of transactions rather than
-	/// blocks. May be called before or after [`update_best_block`] for transactions in the
-	/// corresponding block. See [`update_best_block`] for further calling expectations.
+	/// blocks. May be called before or after [`best_block_updated`] for transactions in the
+	/// corresponding block. See [`best_block_updated`] for further calling expectations.
 	///
 	/// [`block_connected`]: Self::block_connected
-	/// [`update_best_block`]: Self::update_best_block
+	/// [`best_block_updated`]: Self::best_block_updated
```

Comment:
Should we convert these into an `impl chain::Confirm for ChainMonitor`?

Reply:
Yeah, this likely would be needed when implementing a sync utility. Added a separate block with `Send` and `Sync` restrictions akin to `Listen`. Let me know if I'm misunderstanding when this is needed.

Could also move the implementations directly into the added definitions if you'd like. It's unclear to me why they are split for `Listen`.

Reply:
>  Added a separate block with Send and Sync restrictions akin to Listen. Let me know if I'm misunderstanding when this is needed.

Hmm, frankly not sure why those are there, I assume we used to need them for something, but now don't? Either way, it seems to compile without them, so no reason to have extra bounds on it - same goes for the `chain::Listen` impl as well now.

Also, I think if the traits are implemented we should drop the public functions which are duplicative. `block_connected` is only there because its slightly different from the `chain::Listen` `block_connected`, IIRC.

Reply:
Yeah, sounds good. Note that this requires having `full_stack` uses `chain::Confirm` for `ChainMonitor` otherwise it would be a much larger change. FWIW, it is already doing so for `ChannelManager`.